airbyte-integrations/connectors/source-google-analytics-data-api/source_google_analytics_data_api/source.py

```diff
@@ -154,13 +154,21 @@ def get_json_schema(self) -> Mapping[str, Any]:
     def next_page_token(self, response: requests.Response) -> Optional[Mapping[str, Any]]:
         r = response.json()
 
-        if all(key in r for key in ["limit", "offset", "rowCount"]):
-            limit, offset, total_rows = r["limit"], r["offset"], r["rowCount"]
+        if "rowCount" in r:
+            limit = self.config["limit"]
+            total_rows = r["rowCount"]
+            if "next_page_offset" in self.config and self.config["next_page_offset"] is not None:
+                offset = self.config["next_page_offset"]
+            else:
+                offset = self.config["offset"]
 
-            if total_rows <= offset:
+            if total_rows <= offset + limit:
+                self.config["next_page_offset"] = None
                 return None
 
-            return {"limit": limit, "offset": offset + limit}
+            self.config["next_page_offset"] = offset + limit
```

Comment:
Why do we need to keep `next_page_offset` inside `self.config` dict?

The main purpose of this method `def next_page_token` is to pass to next functions like `request_body_json` the value of the token